presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -478,28 +485,75 @@ private void testRoundTripType(Type type, List<?> readValues)
 
     public void testRoundTripTypes(List<Type> types, List<List<?>> readValues, List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters)
             throws Exception
+    {
+        testRoundTripTypes(types, readValues, filters, Optional.empty());
+    }
+
+    public void testRoundTripTypes(List<Type> types, List<List<?>> readValues, List<Map<Integer, Map<Subfield, TupleDomainFilter>>> filters, Optional<List> order)
+            throws Exception
     {
         assertEquals(types.size(), readValues.size());
+        Optional<OrderChecker> optionalOrderChecker = Optional.empty();
+
+        if (order.isPresent()) {
+            OrderChecker orderChecker = new OrderChecker();
+            orderChecker.setDesiredOrder(order.get());
+            // Turn normal TupleDomainFilters into their corresponding OrderTrackingOrcFilters.
```

Comment:
This is too soon. I'd delay changing the filters until it is type to create ORC reader. This way you won't need `start/stop` methods as the filter with tracking will only be used by the reader.